src/test/java/uk/gov/ida/verifyserviceprovider/configuration/EuropeanIdentityConfigurationTest.java

```diff
@@ -0,0 +1,124 @@
+        Certificate europeanConfigCert = europeanIdentityConfiguration.getTrustStore().getCertificate("metadataCA");
+
+        assertThat(europeanIdentityConfiguration.getTrustStore().containsAlias("rootCA")).isTrue();
+        assertThat(europeanIdentityConfiguration.getTrustStore().containsAlias("metadataCA")).isTrue();
+        assertThat(europeanIdentityConfiguration.getTrustStore().size()).isEqualTo(2);
+        assertThat(europeanConfigCert).isEqualTo(testEntryCert);
+    }
+
+    @Test
+    public void shouldReadTrustAnchorAtUriGivenEidasIsEnabledWithHubEnvironmentSetToIntegrationWhenTrustAnchorUriHasbeenSpecified() throws IOException {
+
+        EuropeanIdentityConfiguration europeanIdentityConfiguration = OBJECT_MAPPER.readValue(config, EuropeanIdentityConfiguration.class);
+        europeanIdentityConfiguration.setEnvironment(HubEnvironment.INTEGRATION);
+
+        assertThat(europeanIdentityConfiguration.getTrustAnchorUri()).isEqualTo(URI.create("http://example.com"));
+    }
+
+    @Test
+    public void shouldReadTrustAnchorAtUriGivenEidasIsEnabledWithoutHubEnvironmentSetToIntegrationWhenTrustAnchorUriHasbeenSpecified() throws IOException {
+
+        EuropeanIdentityConfiguration europeanIdentityConfiguration = OBJECT_MAPPER.readValue(config, EuropeanIdentityConfiguration.class);
+
+        assertThat(europeanIdentityConfiguration.getTrustAnchorUri()).isEqualTo(URI.create("http://example.com"));
+    }
+
+    @Test
+    public void shouldReadHubEnvironmentTrustAnchorGivenEidasIsEnabledWithHubEnvironmentSetToIntegrationWhenNoTrustAnchorUriHasbeenSpecified() throws IOException {
+        config = "{" +
+                "\"enabled\":\"true\"," +
+                "\"trustStore\": {" +
+                "\"path\": \"" + keyStore.getAbsolutePath() + "\"," +
+                "\"password\": \"" + keyStore.getPassword() + "\"" +
+                "}}";
+
+        EuropeanIdentityConfiguration europeanIdentityConfiguration = OBJECT_MAPPER.readValue(config, EuropeanIdentityConfiguration.class);
+        europeanIdentityConfiguration.setEnvironment(HubEnvironment.INTEGRATION);
+
+        assertThat(europeanIdentityConfiguration.getTrustAnchorUri()).isNotEqualTo(URI.create("http://example.com"));
+    }
+
+
+    @Test(expected = NullPointerException.class)
+    public void shouldErrrorWhenReadingTrustAnchorGivenEidasIsEnabledWithNoSpecifiedTrustAnchorUriAndNoSpecifiedEnvironment() throws IOException {
+        config = "{" +
+                "\"enabled\":\"true\"," +
+                "\"trustStore\": {" +
+                "\"path\": \"" + keyStore.getAbsolutePath() + "\"," +
+                "\"password\": \"" + keyStore.getPassword() + "\"" +
+                "}}";
+
+        EuropeanIdentityConfiguration europeanIdentityConfiguration = OBJECT_MAPPER.readValue(config, EuropeanIdentityConfiguration.class);
+        europeanIdentityConfiguration.getTrustAnchorUri();
+    }
```

Comment:
The extra tests look good to me; I like that you're checking we get an error when we don't specify anything.

One tiny thing: some of the test names say Hasbeen instead of HasBeen.  Only worth fixing if you're making other changes.